src/common/dav/wal_tx.h

```diff
@@ -0,0 +1,42 @@
+/**
+ * (C) Copyright 2021-2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#ifndef __DAOS_COMMON_DAV_WAL_TX_
+#define __DAOS_COMMON_DAV_WAL_TX_
+
+#include <gurt/list.h>
+#include <daos_types.h>
+#include <daos/mem.h>
+
+struct dav_obj;
+
+struct wal_action {
+	d_list_t                wa_link;
+	struct umem_action      wa_act;
+};
+
+struct dav_tx {
+	struct dav_obj		*wt_dav_hdl;
+	uint64_t		 wt_id;
+	d_list_t		 wt_redo;
```

Comment:
Do we have to convert ulog into a list? Is it possible to iterate the ulog directly? I suppose that could save the overhead of such covert.

Reply:
In this PR the log is a list. We can make it a vector later.